plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/cache/DeltaLakeCacheKeyProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+
+import java.util.Optional;
+
+public class DeltaLakeCacheKeyProvider
+        implements CacheKeyProvider
+{
+    /**
+     * Get the cache key of a TrinoInputFile. Returns Optional.empty() if the file is not cacheable.
+     */
+    @Override
+    public Optional<String> getCacheKey(TrinoInputFile delegate)
+    {
+        // TODO: Enable caching of parquet checkpoint files once the CheckpointEntryIterator is always closed
```

Comment:
Is this TODO still pending ? If it is, please include a link to an open GH issue

Reply:
It should be resolved by https://github.com/trinodb/trino/pull/20054 now. I'll enable the caching of checkpoint files.

Reply:
The `TestDeltaLakeAlluxioCacheFileOperations` does have coverage for the cache operations for checkpoint files, but it's unclear how much coverage on checkpoints we get through the `TestDeltaLakeAlluxioCacheMinioAndHmsConnectorSmokeTest` test. It might be prudent to just leave it as is, and consider it as a future optimization. I'll keep the code as is, and leave a TODO to enable caching of checkpoint files.

Enabling caching of checkpoint files could make the `delta.checkpoint-filtering.enabled=true` feature work better. When we tried enabling this feature earlier we saw a major slow down of the planning and analysis phases. The in-memory cache of checkpoints seemed to no longer be used, and lots of queries were made by the coordinator to object storage to fetch checkpoint files. However, if we only need to read the checkpoint files from disk it could work without the in-memory checkpoint cache.